src/lib/DefaultRuleManager.ts

```diff
@@ -126,6 +133,28 @@ export class DefaultRuleManager implements RuleManager {
 		}
 	}
 
+	protected async emitRunTelemetry(runDescriptorList: ExtendedRunDescriptor[]): Promise<void> {
+		// Telemetry data always specifies an event name.
+		const runTelemetryObject: TelemetryData = {
+			eventName: "ENGINE_EXECUTION"
+		};
+
+		// For engine execution, the telemetry data should include information about which engines ran and which engines
+		// didn't.
+		const executedEngineNames: Set<string> = new Set(runDescriptorList.map(descriptor => descriptor.engine.getName().toLowerCase()));
+		const allEngines: RuleEngine[] = await Controller.getAllEngines();
+		for (const engine of allEngines) {
+			const engineName = engine.getName().toLowerCase();
+			runTelemetryObject[engineName] = executedEngineNames.has(engineName);
+		}
+
+		// The telemetry data also needs to have the SFDX version.
+		runTelemetryObject['sfdxVersion'] = await SfdxUtils.getSfdxVersion();
+
+		// We can use the Lifecycle convenience method to send off our telemetry data.
+		await Lifecycle.getInstance().emitTelemetry(runTelemetryObject);
```

Comment:
The `.emitTelemetry()` method is what we were waiting on Shane and his team to finish.